AfrikaBurn/MapViewController.swift

```diff
@@ -0,0 +1,194 @@
+        commonInit()
+    }
+    
+    func commonInit() {
+        showsScale = true
+        showsCompass = true
+        delegate = self
+        userTrackingMode = .followWithHeading
+        add(overlay, level: .aboveRoads)
+        setVisibleMapRect(BurnMap.boundingMapRect, animated: false)
+    }
+    func mapView(_ mapView: MKMapView, rendererFor overlay: MKOverlay) -> MKOverlayRenderer {
+        switch overlay {
+        case is BurnMapOverlay: return BurnMapOverlayView(overlay: overlay, overlayImage: Resources.mapOverlayImage)
+        default: return MKOverlayRenderer(overlay: overlay)
+        }
+    }
+}
+
+class BurnMapOverlay: NSObject, MKOverlay {
+    let boundingMapRect: MKMapRect = BurnMap.boundingMapRect
+    var coordinate: CLLocationCoordinate2D = CLLocationCoordinate2DMake(19.751633252193361, -32.326217923051772)
+    
+    func canReplaceMapContent() -> Bool {
+        return true
+    }
+}
+
+class BurnMapOverlayView: MKOverlayRenderer {
+    let overlayImage: UIImage
+    
+    init(overlay: MKOverlay, overlayImage: UIImage) {
+        self.overlayImage = overlayImage
+        super.init(overlay: overlay)
+    }
+    
+    override func draw(_ mapRect: MKMapRect, zoomScale: MKZoomScale, in context: CGContext) {
+        guard let imageRef = overlayImage.cgImage else {
+            return
+        }
+        let theMapRect = overlay.boundingMapRect
+        let theRect = rect(for: theMapRect)
+        
+        context.scaleBy(x: 1, y: -1)
+        context.translateBy(x: 0, y: -theRect.height)
+        context.draw(imageRef, in: theRect)
+    }
+}
+
+
+extension MapViewController: MKMapViewDelegate {
+    
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        // If annotation is not of type RestaurantAnnotation (MKUserLocation types for instance), return nil
+        if !(annotation is BurnAnnotation){
+            return nil
+        }
+        
+        var annotationView = self.mapView.dequeueReusableAnnotationView(withIdentifier: "Pin")
+        
+        if annotationView == nil{
+            annotationView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: "Pin")
+            annotationView?.canShowCallout = true
+        }else{
+            annotationView?.annotation = annotation
+        }
+        
+        let burnAnnotation = annotation as! BurnAnnotation
+        
+        let blurb = UILabel(frame: CGRect(x: 0,y: 0,width: annotationView!.frame.size.width,height: 200))
+        blurb.font = UIFont(name: "Helvetica", size: 12)
+        blurb.text = burnAnnotation.element?.shortBlurb
+        blurb.numberOfLines = 6
+        annotationView?.detailCalloutAccessoryView = blurb
+        
+        
+        // Left Accessory
+        // We could make this change the icon based on the element type. Eg. mutant vehicles etc.
+        annotationView?.leftCalloutAccessoryView = UIImageView(image: UIImage(named: "fire.png"))
```

Comment:
I think this is a good approach, and we could use the same approach on the tableview too.